service/controller/azuremachinepool/handler/migration/resource.go

```diff
@@ -0,0 +1,159 @@
+package migration
+
+import (
+	"context"
+	"encoding/json"
+	"reflect"
+
+	oldcapiexpv1alpha3 "github.com/giantswarm/apiextensions/v6/pkg/apis/capiexp/v1alpha3"
+	oldcapzexpv1alpha3 "github.com/giantswarm/apiextensions/v6/pkg/apis/capzexp/v1alpha3"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/giantswarm/operatorkit/v7/pkg/controller/context/reconciliationcanceledcontext"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	capzexp "sigs.k8s.io/cluster-api-provider-azure/exp/api/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/giantswarm/azure-operator/v5/service/controller/key"
+)
+
+const (
+	// Name is the identifier of the resource.
+	Name = "migration"
+)
+
+type Config struct {
+	CtrlClient client.Client
+	Logger     micrologger.Logger
+}
+
+type Resource struct {
+	ctrlClient client.Client
+	logger     micrologger.Logger
+}
+
+func New(c Config) (*Resource, error) {
+	if c.CtrlClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.CtrlClient must not be empty", c)
+	}
+	if c.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", c)
+	}
+
+	r := &Resource{
+		ctrlClient: c.CtrlClient,
+		logger:     c.Logger,
+	}
+
+	return r, nil
+}
+
+func (r Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
```

Comment:
New AzureMachinePool handler that will copy status from the old experimental AzureMachinePool CR to the new AzureMachinePool CR.

Reply:
And then it will delete the old AzureMachinePool CR.

Reply:
Maybe better to add your comment as a code comment in `EnsureCreated`